synfig-studio/src/gui/widgets/widget_link.h

```diff
@@ -40,18 +42,48 @@ namespace studio {
 
 class Widget_Link: public Gtk::ToggleButton
 {
-	synfig::String tooltip_inactive_;
-	synfig::String tooltip_active_;
+	std::string tooltip_inactive_;
+	std::string tooltip_active_;
 
 	Gtk::Image *icon_off_;
 	Gtk::Image *icon_on_;
 
+	void init();
+	void init(const std::string &tlt_inactive, const std::string &tlt_active);
+	void set_up_icons();
+
 protected:
 	void on_toggled();
 
 public:
+	Widget_Link();
 	Widget_Link(const std::string &tlt_inactive, const std::string &tlt_active);
 	~Widget_Link();
+
+// Glade & GtkBuilder related
+private:
+	Glib::RefPtr<Gtk::Builder> builder;
+
+	const bool IS_WIDGET_FROM_BUILDER;
+
+	static GType gtype;
+
+	void refresh_tooltip_text();
+
+public:
+	Widget_Link(BaseObjectType *cobject);
+	Widget_Link(
+			BaseObjectType *cobject, const Glib::RefPtr<Gtk::Builder> &builder,
+			const std::string tlt_inactive, const std::string tlt_active);
```

Comment:
Why this line break?
Anyway, I believe this constructor is redundant, if you use `Glib::PropertyProxy<std::string> property_tooltip_active` properly. The members `std::string tooltip_inactive_` and `std::string tooltip_inactive_` would be unnecessary btw.

Check out how I used them in https://github.com/synfig/synfig/pull/2095/files  (ColorSlider file, for example)


https://user-images.githubusercontent.com/1323987/113710283-e630b500-96b9-11eb-9f9c-bae0571e9d54.mov

Reply:
Working on this now.

Give me some time. This is still new to me :P

Thanks for the links 👍🏽 👍🏽 👍🏽   Looks like I only need to define two new property ids on synfig-catalog.xml and call `default` for proxy property inside the constructor instead of empty string `""`

Reply:
You defined the `default_color` variable by code
```c++
static const Gdk::RGBA default_color("#F00");

...
Widget_ColorSlider::Widget_ColorSlider(Type x)
    : ...
    property_color(*this, "color", default_color),
...
```

I thought our custom widget property `tooltip_active`/`tooltip_inactive` "default" values could be called (???) since they are defined in synfig-catalog.xml file
```xml
<glade-widget-class name="gtkmm__CustomObject_widget_link"  ... >
  <properties>
    <property id="tooltip-active" default="Link width and height" />
    <property id="tooltip-inactive" default="Unlink width and height" />
  </properties>
</glade-widget-class>
```

Just out of curiosity I did try this
```c++
// big mistake I made here without instantiating object first
value_active(*this, "tooltip-active", property_tooltip_active().get_value()),
```
I'm a little lost here. Is there a way to get those values set on the catalog file?

Reply:
Sorry I missed your answer/question here.
The catalog is used only for Glade editor, so the widget c++ code doesn't get the the default value provided in this .xml, that isn't even bundled in Synfig Studio.

Therefore:
`value_active(*this, "tooltip-active", property_tooltip_active().get_value()),` doesn't make sense.
First, the property doesn't have a default value loaded from the .xml.

Second: if you use the PropertyProxy, the respective/correspondent Property should be private.
The (property/signal/etc)Proxy is to forbid some actions on the (property/signal) by other classes.
In the PropertyProxy case, specifically, it also allows to access a signal_changed(), triggered when the property changes (by, for example, `property_x = 3;`

Reply:
> The catalog is used only for Glade editor,

Makes complete sense :)

Anyway, I remember this custom widget property was your suggestion from [this PR](https://github.com/synfig/synfig/pull/1577#discussion_r459529480) and that's what you're trying to point to I think. Just pass the string directly: `"Link"` `"Unlink"`

And over here my focus was to easily migrate below lines to future glade UI file by passing them (strings) as arguments
https://github.com/synfig/synfig/blob/8fdfae6877260c265ffb8a60567740441f77d413/synfig-studio/src/gui/renddesc.cpp#L596-L597

Anyway, my PR seems to break older system because `get_widget_derived(..., arg3, arg4)` isn't supported by old-old stable system and thus this widget will be built without the tooltip messages.

Proceeding with your suggested changes...

And thanks for this wonderful post https://gist.github.com/rodolforg/802222eb41287da744646529615d91e2
This time I got the purpose LOL